git/git.go

```diff
@@ -19,14 +19,22 @@ type Git struct {
 
 	DeletedBranches      []string
 	BranchDeletionErrors []branchDeletionError
+
+	directory string
+}
+
+// NewExecutor Constructor for git executor
+func NewExecutor(directory string) Git {
```

Comment:
I actually tried returning a `*Git` first, and the compiler complained.

The way I have come to understand this is that `NewExecutor` returns a `Git`, because it is creating a new instance that isn't yet being referenced anywhere? (this may be flawed reasoning)